Compiling error: variable binding in a condition requires an initializer

swift --version
swift-driver version: 1.45.2 Apple Swift version 5.6.1 (swiftlang-5.6.0.323.66 clang-1316.0.20.12)
Target: x86_64-apple-macosx12.0


....../RSCore/Sources/RSCore/CloudKit/CloudKitZone.swift:767:18: error: variable binding in a condition requires an initializer
                                if let error {
                                             ^
....../RSCore/Sources/RSCore/CloudKit/CloudKitZone.swift:788:19: error: variable binding in a condition requires an initializer
                                        if let error {
                                                     ^

--- a/Sources/RSCore/CloudKit/CloudKitZone.swift
+++ b/Sources/RSCore/CloudKit/CloudKitZone.swift
@@ -764,17 +764,10 @@ public extension CloudKitZone {
 			guard incremental else { return }
 			
 			wasChanged(updated: updatedRecords, deleted: deletedRecordKeys, token: token) { error in
-                #if swift(>=5.7)
-				if let error {
-					op.cancel()
-					completion(.failure(error))
-				}
-                #else
                 if let error = error {
                     op.cancel()
                     completion(.failure(error))
                 }
-                #endif
 			}
 			updatedRecords = [CKRecord]()
 			deletedRecordKeys = [CloudKitRecordKey]()
@@ -792,17 +785,10 @@ public extension CloudKitZone {
         op.recordZoneFetchCompletionBlock = { zoneID ,token, _, finalChange, error in
 			if case .success = CloudKitZoneResult.resolve(error) {
 				wasChanged(updated: updatedRecords, deleted: deletedRecordKeys, token: token) { error in
-                    #if swift(>=5.7)
-					if let error {
-						op.cancel()
-						completion(.failure(error))
-					}
-                    #else
                     if let error = error {
                         op.cancel()
                         completion(.failure(error))
                     }
-                    #endif
 				}
 			}
 			updatedRecords = [CKRecord]()

--- a/Sources/RSCore/CloudKit/CloudKitZone.swift
+++ b/Sources/RSCore/CloudKit/CloudKitZone.swift
@@ -764,10 +764,17 @@ public extension CloudKitZone {
 			guard incremental else { return }
 			
 			wasChanged(updated: updatedRecords, deleted: deletedRecordKeys, token: token) { error in
+                #if swift(>=5.7)
 				if let error {
 					op.cancel()
 					completion(.failure(error))
 				}
+                #else
+                if let error = error {
+                    op.cancel()
+                    completion(.failure(error))
+                }
+                #endif
 			}
 			updatedRecords = [CKRecord]()
 			deletedRecordKeys = [CloudKitRecordKey]()
@@ -785,10 +792,17 @@ public extension CloudKitZone {
         op.recordZoneFetchCompletionBlock = { zoneID ,token, _, finalChange, error in
 			if case .success = CloudKitZoneResult.resolve(error) {
 				wasChanged(updated: updatedRecords, deleted: deletedRecordKeys, token: token) { error in
+                    #if swift(>=5.7)
 					if let error {
 						op.cancel()
 						completion(.failure(error))
 					}
+                    #else
+                    if let error = error {
+                        op.cancel()
+                        completion(.failure(error))
+                    }
+                    #endif
 				}
 			}
 			updatedRecords = [CKRecord]()

